### ISSUE TITLE
Pinpoint Agent does not connect to collector and web in RedHat [SOLVED]

### ISSUE THREAD

**Issue by ing-arriola** [naver/pinpoint#4130]
#### Description
I have been doing tests with Pinpoint on Ubuntu Server 16.04 and I have not had any problems, then go on to perform the same procedure as in Ubuntu on Server RedHat 6.5 to which I have installed version 8 of Java and all the components that on Ubuntu they worked; After that I verified that the Pinpoint web interface works correctly and if it does it like Hbase and Hadoop; The problem arises when I have deployed the Agent in a WAS WebLogic since the agent is not able to connect to the Hbase and therefore does not register and then see their data in the Pinpoint Web Interface. I have disabled the Redhat Firewall and I have even disabled SeELinux but the agent still does not connect. Hoping to receive help regarding the aforementioned problem, I attached the Logs of both the agent and the Tomcat where I deployed Pinpoint.

# **_FROM CATALINA.OUT_**

2018-05-16 10:19:27 [DEBUG](c.n.p.c.h.AgentEventHandler        :80 ) Received TAgentStatBatch=TAgentStatBatch(agentId:agenteopsbasicas, startTimestamp:1526487145522, agentStats:[TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487542738, collectInterval:4999, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123286944, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.002025658338960162, systemCpuLoad:0.016205266711681297), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0)), TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487547737, collectInterval:4999, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123286944, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.002704530087897228, systemCpuLoad:0.006085192697768763), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0)), TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487552737, collectInterval:5000, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123286944, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.002027027027027027, systemCpuLoad:0.008108108108108109), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0)), TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487557737, collectInterval:5000, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123289632, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.002702702702702703, systemCpuLoad:0.00472972972972973), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0)), TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487562736, collectInterval:4999, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123289632, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.002025658338960162, systemCpuLoad:0.014179608372721135), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0)), TAgentStat(agentId:agenteopsbasicas, startTimestamp:1526487145522, timestamp:1526487567741, collectInterval:5005, gc:TJvmGc(type:PARALLEL, jvmMemoryHeapUsed:220775792, jvmMemoryHeapMax:1014497280, jvmMemoryNonHeapUsed:123294240, jvmMemoryNonHeapMax:-1, jvmGcOldCount:4, jvmGcOldTime:261), cpuLoad:TCpuLoad(jvmCpuLoad:0.0027100271002710027, systemCpuLoad:0.005420054200542005), transaction:TTransaction(sampledNewCount:0, sampledContinuationCount:0, unsampledNewCount:0, unsampledContinuationCount:0), activeTrace:TActiveTrace(histogram:TActiveTraceHistogram(version:0, histogramSchemaType:2, activeTraceCount:[0, 0, 0, 0])), dataSourceList:TDataSourceList(dataSourceList:[]), responseTime:TResponseTime(), fileDescriptor:TFileDescriptor(openFileDescriptorCount:379), directBuffer:TDirectBuffer(directCount:12, directMemoryUsed:113868, mappedCount:0, mappedMemoryUsed:0))])
2018-05-16 10:19:29 [DEBUG](c.n.p.c.h.SpanHandler              :76 ) Received SPAN=TSpan(agentId:agenteopsbasicas, applicationName:opbasicasjsp, agentStartTime:1526487145522, transactionId:00 01 B2 B8 E5 CD B6 2C 08, spanId:-5085017144351328408, startTime:1526487569593, elapsed:3, rpc:/bea_wls_deployment_internal/DeploymentService, serviceType:1070, endPoint:172.23.112.192:7004, remoteAddr:172.23.112.192, flag:0, err:0, spanEventList:[TSpanEvent(sequence:0, startElapsed:1, endElapsed:2, serviceType:1071, depth:1, apiId:1)], apiId:-2, applicationServiceType:1070, loggingTransactionInfo:0)
2018-05-16 10:19:29 [ERROR](o.a.h.h.c.AsyncProcess             :420) Failed to get region location 
org.apache.hadoop.hbase.TableNotFoundException: TraceV2
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.locateRegionInMeta(ConnectionManager.java:1283)
	at org.apache.hadoop.hbase.client.ConnectionManager$HConnectionImplementation.locateRegion(ConnectionManager.java:1181)
	at org.apache.hadoop.hbase.client.AsyncProcess.submit(AsyncProcess.java:410)
	at org.apache.hadoop.hbase.client.AsyncProcess.submit(AsyncProcess.java:359)
	at org.apache.hadoop.hbase.client.BufferedMutatorImpl.backgroundFlushCommits(BufferedMutatorImpl.java:238)
	at org.apache.hadoop.hbase.client.BufferedMutatorImpl.flush(BufferedMutatorImpl.java:190)
	at org.apache.hadoop.hbase.client.HTable.flushCommits(HTable.java:1434)
	at org.apache.hadoop.hbase.client.HTable.put(HTable.java:1018)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2$8.doInTable(HbaseTemplate2.java:353)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.execute(HbaseTemplate2.java:766)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.put(HbaseTemplate2.java:350)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.asyncPut(HbaseTemplate2.java:400)
	at com.navercorp.pinpoint.collector.dao.hbase.HbaseTraceDaoV2.insert(HbaseTraceDaoV2.java:67)
	at com.navercorp.pinpoint.collector.handler.SpanHandler.handleSimple(SpanHandler.java:82)
	at com.navercorp.pinpoint.collector.receiver.AbstractDispatchHandler.dispatchSendMessage(AbstractDispatchHandler.java:58)
	at com.navercorp.pinpoint.collector.receiver.DispatchHandlerWrapper.dispatchSendMessage(DispatchHandlerWrapper.java:47)
	at com.navercorp.pinpoint.collector.receiver.udp.BaseUDPHandlerFactory$DispatchPacket.receive(BaseUDPHandlerFactory.java:89)
	at com.navercorp.pinpoint.collector.receiver.udp.BaseUDPHandlerFactory$DispatchPacket.receive(BaseUDPHandlerFactory.java:68)
	at com.navercorp.pinpoint.collector.receiver.udp.Task.run(Task.java:44)
	at com.navercorp.pinpoint.collector.monitor.MonitoredRunnableDecorator$InstrumentedRunnable.run(MonitoredRunnableDecorator.java:113)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2018-05-16 10:19:29 [WARN ](c.n.p.c.h.SpanHandler              :90 ) Span handle error. Caused:Failed 1 action: TraceV2: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: TraceV2: 1 time, . Span:TSpan(agentId:agenteopsbasicas, applicationName:opbasicasjsp, agentStartTime:1526487145522, transactionId:00 01 B2 B8 E5 CD B6 2C 08, spanId:-5085017144351328408, startTime:1526487569593, elapsed:3, rpc:/bea_wls_deployment_internal/DeploymentService, serviceType:1070, endPoint:172.23.112.192:7004, remoteAddr:172.23.112.192, flag:0, err:0, spanEventList:[TSpanEvent(sequence:0, startElapsed:1, endElapsed:2, serviceType:1071, depth:1, apiId:1)], apiId:-2, applicationServiceType:1070, loggingTransactionInfo:0)
com.navercorp.pinpoint.common.hbase.HbaseSystemException: Failed 1 action: TraceV2: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: TraceV2: 1 time, 
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.execute(HbaseTemplate2.java:775)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.put(HbaseTemplate2.java:350)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.asyncPut(HbaseTemplate2.java:400)
	at com.navercorp.pinpoint.collector.dao.hbase.HbaseTraceDaoV2.insert(HbaseTraceDaoV2.java:67)
	at com.navercorp.pinpoint.collector.handler.SpanHandler.handleSimple(SpanHandler.java:82)
	at com.navercorp.pinpoint.collector.receiver.AbstractDispatchHandler.dispatchSendMessage(AbstractDispatchHandler.java:58)
	at com.navercorp.pinpoint.collector.receiver.DispatchHandlerWrapper.dispatchSendMessage(DispatchHandlerWrapper.java:47)
	at com.navercorp.pinpoint.collector.receiver.udp.BaseUDPHandlerFactory$DispatchPacket.receive(BaseUDPHandlerFactory.java:89)
	at com.navercorp.pinpoint.collector.receiver.udp.BaseUDPHandlerFactory$DispatchPacket.receive(BaseUDPHandlerFactory.java:68)
	at com.navercorp.pinpoint.collector.receiver.udp.Task.run(Task.java:44)
	at com.navercorp.pinpoint.collector.monitor.MonitoredRunnableDecorator$InstrumentedRunnable.run(MonitoredRunnableDecorator.java:113)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: TraceV2: 1 time, 
	at org.apache.hadoop.hbase.client.AsyncProcess$BatchErrors.makeException(AsyncProcess.java:258)
	at org.apache.hadoop.hbase.client.AsyncProcess$BatchErrors.access$2000(AsyncProcess.java:238)
	at org.apache.hadoop.hbase.client.AsyncProcess.waitForAllPreviousOpsAndReset(AsyncProcess.java:1817)
	at org.apache.hadoop.hbase.client.BufferedMutatorImpl.backgroundFlushCommits(BufferedMutatorImpl.java:240)
	at org.apache.hadoop.hbase.client.BufferedMutatorImpl.flush(BufferedMutatorImpl.java:190)
	at org.apache.hadoop.hbase.client.HTable.flushCommits(HTable.java:1434)
	at org.apache.hadoop.hbase.client.HTable.put(HTable.java:1018)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2$8.doInTable(HbaseTemplate2.java:353)
	at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.execute(HbaseTemplate2.java:766)

# **_FROM AGENT LOG:_**

2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=101, name='activemq.broker.address}
2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=102, name='activemq.message}
2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=130, name='rabbitmq.exchange}
2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=131, name='rabbitmq.routingkey}
2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=132, name='rabbitmq.properties}
2018-05-16 10:12:27 [INFO ](DefaultAnnotationKeyRegistryService:89 ) add Plugin AnnotationKey:AnnotationKey{code=133, name='rabbitmq.body}
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:106) spanStatClientFactory:null
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:109) spanDataSender:com.navercorp.pinpoint.profiler.sender.UdpDataSender@50687efb
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:112) statDataSender:com.navercorp.pinpoint.profiler.sender.UdpDataSender@517bd097
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:115) clientFactory:com.navercorp.pinpoint.rpc.client.DefaultPinpointClientFactory@142eef62
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:118) tcpDataSender:com.navercorp.pinpoint.profiler.sender.TcpDataSender@4a9cc6cb
2018-05-16 10:12:27 [INFO ](n.p.p.c.m.DefaultApplicationContext:123) agentInformation:AgentInformation{agentId='agenteopsbasicas', applicationName='opbasicasjsp', startTime=1526487145522, pid=9167, machineName='ingarriola-VirtualBox', hostIp='fe80:0:0:0:8508:a4d9:a8cc:a4c3%enp0s3', serverType=WEBLOGIC, jvmVersion='1.8.0_162', agentVersion='1.8.0-SNAPSHOT'}
2018-05-16 10:12:27 [INFO ](c.n.p.p.DefaultAgent               :147) Starting pinpoint Agent.
2018-05-16 10:12:27 [INFO ](.r.DefaultInterceptorRegistryBinder:50 ) bind:DefaultRegistry-0
2018-05-16 10:12:27 [INFO ](c.n.p.p.m.DefaultDeadlockMonitor   :38 ) DefaultDeadlockMonitor started
2018-05-16 10:12:27 [INFO ](c.n.p.p.m.DefaultAgentStatMonitor  :112) AgentStat monitor started
2018-05-16 10:12:27 [INFO ](c.n.p.p.AgentInfoSender            :175) Sending AgentInfo TAgentInfo(hostname:ingarriola-VirtualBox, ip:fe80:0:0:0:8508:a4d9:a8cc:a4c3%enp0s3, ports:, agentId:agenteopsbasicas, applicationName:opbasicasjsp, serviceType:1070, pid:9167, agentVersion:1.8.0-SNAPSHOT, vmVersion:1.8.0_162, startTimestamp:1526487145522, serverMetaData:TServerMetaData(serverInfo:, vmArgs:[-Dweblogic.Name=test3, -Dbea.home=/home/ingarriola/Oracle/Middleware_1036, -Djava.security.policy=/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/lib/weblogic.policy, -Dweblogic.management.server=http://172.23.112.192:7001, -Djava.library.path=/home/ingarriola/Oracle/Middleware_1036/patch_wls1036/profiles/default/native:/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/native/linux/x86_64:/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/native/linux/x86_64/oci920_8:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib, -Dweblogic.system.BootIdentityFile=/home/ingarriola/Oracle/Middleware_1036/user_projects/domains/my_domain/servers/test3/data/nodemanager/boot.properties, -Dweblogic.nodemanager.ServiceEnabled=true, -Dweblogic.security.SSL.ignoreHostnameVerification=false, -Dweblogic.ReverseDNSAllowed=false, -javaagent:/home/ingarriola/Documents/agente1.8/pinpoint-bootstrap-1.8.0-SNAPSHOT.jar, -Dpinpoint.agentId=agenteopsbasicas, -Dpinpoint.applicationName=opbasicasjsp], serviceInfos:[]), jvmInfo:TJvmInfo(version:0, vmVersion:1.8.0_162, gcType:PARALLEL))
2018-05-16 10:12:27 [WARN ](c.n.p.p.AgentInfoSender            :211) request unsuccessful. Cause : Failed 1 action: AgentInfo: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: AgentInfo: 1 time, 
2018-05-16 10:12:33 [INFO ](c.n.p.p.w.WeblogicPlugin           :334) Weblogic Pinpoint Plugin Transform
2018-05-16 10:12:33 [INFO ](c.n.p.p.s.TcpDataSender@Default    :244) request fail. request:TApiMetaData(agentId:agenteopsbasicas, agentStartTime:1526487145522, apiId:1, apiInfo:weblogic.servlet.internal.WebAppServletContext.execute(weblogic.servlet.internal.ServletRequestImpl ServletRequestImpl, weblogic.servlet.internal.ServletResponseImpl ServletResponseImpl), line:2131, type:0) Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:33 [INFO ](c.n.p.p.s.TcpDataSender@Default    :244) request fail. request:TApiMetaData(agentId:agenteopsbasicas, agentStartTime:1526487145522, apiId:-2, apiInfo:WebLogic Servlet Process, line:-1, type:100) Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:37 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=2, maxRetryCount=3, bytes=62, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:37 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=2, maxRetryCount=3, bytes=226, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:37 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=2, maxRetryCount=3, bytes=63, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:47 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=3, maxRetryCount=3, bytes=62, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:47 [WARN ](c.n.p.p.s.RetryQueue               :55 ) discard retry message(RetryMessage{retryCount=3, maxRetryCount=3, bytes=62, messageDescription='TApiMetaData'}).
2018-05-16 10:12:47 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=3, maxRetryCount=3, bytes=63, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:47 [WARN ](c.n.p.p.s.RetryQueue               :55 ) discard retry message(RetryMessage{retryCount=3, maxRetryCount=3, bytes=63, messageDescription='TApiMetaData'}).
2018-05-16 10:12:47 [INFO ](c.n.p.p.s.TcpDataSender@Default    :279) request fail. request:RetryMessage{retryCount=3, maxRetryCount=3, bytes=226, messageDescription='TApiMetaData'}, Caused:Failed 1 action: ApiMetaData: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: ApiMetaData: 1 time, 
2018-05-16 10:12:47 [WARN ](c.n.p.p.s.RetryQueue               :55 ) discard retry message(RetryMessage{retryCount=3, maxRetryCount=3, bytes=226, messageDescription='TApiMetaData'}).
2018-05-16 10:13:27 [INFO ](c.n.p.r.c.PinpointClientHandshaker :312) PinpointClientHandshaker@7a791b66 HandshakeJob completed.
2018-05-16 10:17:27 [INFO ](c.n.p.p.AgentInfoSender            :175) Sending AgentInfo TAgentInfo(hostname:ingarriola-VirtualBox, ip:fe80:0:0:0:8508:a4d9:a8cc:a4c3%enp0s3, ports:, agentId:agenteopsbasicas, applicationName:opbasicasjsp, serviceType:1070, pid:9167, agentVersion:1.8.0-SNAPSHOT, vmVersion:1.8.0_162, startTimestamp:1526487145522, serverMetaData:TServerMetaData(serverInfo:, vmArgs:[-Dweblogic.Name=test3, -Dbea.home=/home/ingarriola/Oracle/Middleware_1036, -Djava.security.policy=/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/lib/weblogic.policy, -Dweblogic.management.server=http://172.23.112.192:7001, -Djava.library.path=/home/ingarriola/Oracle/Middleware_1036/patch_wls1036/profiles/default/native:/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/native/linux/x86_64:/home/ingarriola/Oracle/Middleware_1036/wlserver_10.3/server/native/linux/x86_64/oci920_8:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib, -Dweblogic.system.BootIdentityFile=/home/ingarriola/Oracle/Middleware_1036/user_projects/domains/my_domain/servers/test3/data/nodemanager/boot.properties, -Dweblogic.nodemanager.ServiceEnabled=true, -Dweblogic.security.SSL.ignoreHostnameVerification=false, -Dweblogic.ReverseDNSAllowed=false, -javaagent:/home/ingarriola/Documents/agente1.8/pinpoint-bootstrap-1.8.0-SNAPSHOT.jar, -Dpinpoint.agentId=agenteopsbasicas, -Dpinpoint.applicationName=opbasicasjsp], serviceInfos:[]), jvmInfo:TJvmInfo(version:0, vmVersion:1.8.0_162, gcType:PARALLEL))
2018-05-16 10:17:27 [WARN ](c.n.p.p.AgentInfoSender            :211) request unsuccessful. Cause : Failed 1 action: AgentInfo: 1 time, ; nested exception is org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException: Failed 1 action: AgentInfo: 1 time, 

In advance thank you very much for the help

#### Environment

* Pinpoint version:1.8.0 SNAPSHOT
* OS, WAS, affected library, etc: RedHat 6.5, Hadoop 2.7.2, Hbase 1.2.2, JDK 1.8.0_162
* Additional details (e.g. how your collector/web is set up):Deploy on RedHat 6.5 and Tomcat 8
[agenteopsbasicas-pinpoint.log](https://github.com/naver/pinpoint/files/2010096/agenteopsbasicas-pinpoint.log)

**Comment by minwoo-jung** [comment]
Hi ing-arriola  

**I think agent communicate collector. **
Below log is collector log. collector server receive span data of agent.
```
2018-05-16 10:19:27 [DEBUG](c.n.p.c.h.AgentEventHandler :80 ) ~~
``` 

I think you should read this log
```
2018-05-16 10:19:29 [ERROR](o.a.h.h.c.AsyncProcess :420) Failed to get region location
 org.apache.hadoop.hbase.TableNotFoundException: TraceV2
```

**Comment by ing-arriola** [comment]
Hi @minwoo-jung 

You were right, I have reviewed the Log line that you mention in your comment:
**2018-05-16 10:19:29 [ERROR](o.a.h.h.c.AsyncProcess :420) Failed to get region location
 org.apache.hadoop.hbase.TableNotFoundException: TraceV2**

Additionally there is another person who had a similar problem (View https://github.com/naver/pinpoint/issues/2115), what he did was create the table that requested the error in my case I managed to solve by running the HBASE script that is available in:

In conclusion the problem that I had was due to omit the creation of the tables that is achieved using the aforementioned script.
https://github.com/naver/pinpoint/tree/master/hbase/scripts

Thank you very much for the help!!! =D

**Comment by minwoo-jung** [comment]
@ing-arriola 
I am glad to solved it.
